runtime/lua/vim/_editor.lua

```diff
@@ -106,27 +106,29 @@ vim.log = {
 ---     and closed. Defaults to `false`.
 ---   - stdout: (boolean|function)
 ---     Handle output from stdout. When passed as a function must have the signature `fun(err: string, data: string)`.
----     Defaults to `true`
+---     Defaults to `true`.
 ---   - stderr: (boolean|function)
 ---     Handle output from stderr. When passed as a function must have the signature `fun(err: string, data: string)`.
 ---     Defaults to `true`.
 ---   - text: (boolean) Handle stdout and stderr as text. Replaces `\r\n` with `\n`.
----   - timeout: (integer) Run the command with a time limit. Upon timeout the process is sent the
----     TERM signal (15) and the exit code is set to 124.
+---   - timeout: (integer) Run the command with a time limit (given in milliseconds).
+---     Upon timeout the process is sent the TERM signal (15) and the exit code is set to 124.
 ---   - detach: (boolean) If true, spawn the child process in a detached state - this will make it
 ---     a process group leader, and will effectively enable the child to keep running after the
 ---     parent exits. Note that the child process will still keep the parent's event loop alive
 ---     unless the parent process calls |uv.unref()| on the child's process handle.
 ---
---- @param on_exit (function|nil) Called when subprocess exits. When provided, the command runs
----   asynchronously. Receives SystemCompleted object, see return of SystemObj:wait().
+--- @param on_exit fun(vim.SystemCompleted)|nil Called when subprocess exits. When provided, the
+---           command runs asynchronously. Receives vim.SystemCompleted object, see return of
+---           SystemObj:wait().
 ---
 --- @return vim.SystemObj Object with the fields:
 ---   - pid (integer) Process ID
----   - wait (fun(timeout: integer|nil): SystemCompleted) Wait for the process to complete. Upon
----     timeout the process is sent the KILL signal (9) and the exit code is set to 124. Cannot
----     be called in |api-fast|.
----     - SystemCompleted is an object with the fields:
+---   - wait (fun(timeout: integer|nil): vim.SystemCompleted) Wait for the process to complete,
+---     for {timeout} milliseconds (if {timeout} is given) or the timeout specified in {opts}.
+---     Upon timeout the process is sent the KILL signal (9) and the exit code is set to 124.
```

Comment:
The current behavior can be a bit confusing. Upon timeout w.r.t the `{timeout}` in `:wait(timeout)` signal is SIGKILL (9), and upon timeout w.r.t. the timeout given `opts = { timeout = ... }` signal is SIGKILL (15). Is this by design? How about we use the same signal (SIGTERM)? /cc @lewis6991

Reply:
Since there appears to be no obvious and strong reason to use SIGKILL, I propose we always return `SIGTERM` (as Lewis has agreed) for the sake of consistency. Please see the latest commit.